aws/resource_aws_sns_topic_subscription_test.go

```diff
@@ -31,6 +31,27 @@ func TestAccAWSSNSTopicSubscription_basic(t *testing.T) {
 	})
 }
 
+func TestAccAWSSNSTopicSubscription_withFilterPolicy(t *testing.T) {
+	ri := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSSNSTopicSubscriptionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSSNSTopicSubscriptionConfigWithFilterPolicy(ri),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSSNSTopicExists("aws_sns_topic.test_topic"),
+					testAccCheckAWSSNSTopicSubscriptionExists("aws_sns_topic_subscription.test_subscription"),
+					resource.TestCheckResourceAttr("aws_sns_topic_subscription.test_subscription", "filter_policy",
+						"{\"store\": [\"example_corp\"], \"event\": [\"order_placed\"]}"),
+				),
+			},
+		},
+	})
+}
+
```

Comment:
I'm afraid I cannot test `delivery_policy` in the subscription. It should be easy, just need to add the attribute in one of the HTTP endpoint test, something like

`delivery_policy        = "{\"healthyRetryPolicy\": {\"numRetries\": 51}}"`

and then assert in the acceptance test:

```
resource.TestCheckResourceAttr("aws_sns_topic_subscription.test_subscription", "delivery_policy", "{\"healthyRetryPolicy\": {\"numRetries\": 51}}"),
```